libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -358,3 +368,19 @@ func makeMeta(beat beat.Info) common.MapStr {
 		"uuid":    beat.ID,
 	}
 }
+
+func getClusterUUID() string {
+	stateRegistry := monitoring.GetNamespace("state").GetRegistry()
+	outputsRegistry := stateRegistry.GetRegistry("outputs")
+	if outputsRegistry == nil {
+		return ""
+	}
+
+	elasticsearchRegistry := outputsRegistry.GetRegistry("elasticsearch")
+	if elasticsearchRegistry == nil {
+		return ""
+	}
+
+	snapshot := monitoring.CollectFlatSnapshot(elasticsearchRegistry, monitoring.Full, false)
+	return snapshot.Strings["cluster_uuid"]
```

Comment:
Is `cluster_uuid` always set?

Reply:
No, this is not guaranteed. It will be never set if the beat is not using the `elasticsearch` output OR if it is using the `elasticsearch` output but the initial connection has not yet been established.

Is your concern that `snapshot.Strings["cluster_uuid"]` might fail somehow? If so, this should not happen as `snapshot.Strings` is a `map[string]string` and if the map key `cluster_uuid` doesn't exist, the zero value of `string`, that is `""`, will be returned here.

Reply:
yes, my concern was that it might panic if you try to access a key that not exist. If that is no tthe case, all fine.